[BUG] Integration test failures with DecimalType on Spark-3.4 SNAPSHOT build

**Describe the bug**
Integration tests are failing with DecimalType on Spark-3.4 build.
Pasting subset of failures(covers all tests).
```
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(18,0)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 2048 GPU: 0
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(20,2)] - AssertionError: GPU and CPU decimal values are different at [0, '(a + b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(30,2)] - AssertionError: GPU and CPU decimal values are different at [5, '(a + b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(36,5)] - AssertionError: GPU and CPU decimal values are different at [5, '(a + b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(38,10)] - AssertionError: GPU and CPU decimal values are different at [0, '(a + CAST(100 AS DECIMAL(38,10)))']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition[Decimal(36,-5)] - AssertionError: GPU and CPU decimal values are different at [5, '(a + b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_addition_ansi_no_overflow[Decimal(19,0)] - AssertionError: GPU and CPU decimal values are different at [0, '(a + b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(18,0)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 2048 GPU: 0
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(20,2)] - AssertionError: GPU and CPU decimal values are different at [8, '(a - b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(30,2)] - AssertionError: GPU and CPU decimal values are different at [1, '(a - b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(36,5)] - AssertionError: GPU and CPU decimal values are different at [1, '(a - b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(38,10)] - AssertionError: GPU and CPU decimal values are different at [0, '(a - CAST(100 AS DECIMAL(38,10)))']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction[Decimal(36,-5)] - AssertionError: GPU and CPU decimal values are different at [1, '(a - b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_subtraction_ansi_no_overflow[Decimal(19,0)] - AssertionError: GPU and CPU decimal values are different at [2, '(a - b)']
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication[Decimal(7,-3)] - py4j.protocol.Py4JJavaError: An error occurred while calling o17043.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication[Decimal(38,10)] - py4j.protocol.Py4JJavaError: An error occurred while calling o18609.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication[Decimal(38,-10)] - py4j.protocol.Py4JJavaError: An error occurred while calling o18870.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_ansi_enabled[Decimal(7,3)] - py4j.protocol.Py4JJavaError: An error occurred while calling o20182.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_ansi_enabled[Decimal(19,0)] - py4j.protocol.Py4JJavaError: An error occurred while calling o20381.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_mixed[Byte-Decimal(6,5)] - py4j.protocol.Py4JJavaError: An error occurred while calling o21441.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_mixed[Byte-Decimal(30,10)] - py4j.protocol.Py4JJavaError: An error occurred while calling o22857.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_mixed[Short-Decimal(6,5)] - py4j.protocol.Py4JJavaError: An error occurred while calling o23742.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_multiplication_mixed[Short-Decimal(6,4)] - py4j.protocol.Py4JJavaError: An error occurred while calling o23919.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division[Decimal(7,-3)] - py4j.protocol.Py4JJavaError: An error occurred while calling o47443.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division[Decimal(6,3)] - py4j.protocol.Py4JJavaError: An error occurred while calling o47700.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division[Decimal(5,5)] - py4j.protocol.Py4JJavaError: An error occurred while calling o47957.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed[Decimal(5,3)-Decimal(4,1)] - py4j.protocol.Py4JJavaError: An error occurred while calling o56967.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_add_overflow_with_ansi_enabled[[Decimal('-99999999999999999999999999999999999999')]-DecimalType(38,0)-a + -1] - pyspark.sql.utils.ParseException: 
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed[Decimal(5,3)-Decimal(5,0)] - py4j.protocol.Py4JJavaError: An error occurred while calling o57148.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed[Decimal(5,3)-Decimal(5,1)] - py4j.protocol.Py4JJavaError: An error occurred while calling o57329.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_add_overflow_with_ansi_enabled[[Decimal('99999999999999999999999999999999999999')]-DecimalType(38,0)-a + 10] - pyspark.sql.utils.ParseException: 
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed_larger_dec[Decimal(27,7)-Decimal(28,18)] - py4j.protocol.Py4JJavaError: An error occurred while calling o62216.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_day_time_interval_division_overflow[LongType()-[datetime.timedelta(days=-106751992, seconds=71945, microseconds=224192), -1]] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_day_time_interval_division_overflow[IntegerType()-[datetime.timedelta(days=-106751992, seconds=71945, microseconds=224192), -1]] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed_larger_dec[Decimal(20,-3)-Decimal(30,10)] - py4j.protocol.Py4JJavaError: An error occurred while calling o62397.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_division_mixed_larger_dec[Decimal(20,-3)-Decimal(28,18)] - py4j.protocol.Py4JJavaError: An error occurred while calling o62578.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_special_decimal_division - py4j.protocol.Py4JJavaError: An error occurred while calling o62755.collectToPython.
FAILED ../../src/main/python/arithmetic_ops_test.py::test_day_time_interval_divided_by_zero[ByteType()-[datetime.timedelta(seconds=1), 0]] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_day_time_interval_divided_by_zero[ShortType()-[datetime.timedelta(seconds=1), 0]] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_mod_pmod_by_zero[pmod(a, cast(0 as {}))-Decimal(7,3)] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_mod_pmod_by_zero[pmod(a, cast(0 as {}))-Decimal(12,2)] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_day_time_interval_scalar_divided_by_zero[DoubleType()--0.0] - AssertionError
FAILED ../../src/main/python/arithmetic_ops_test.py::test_int_division_mixed[Decimal(13,2)-Decimal(6,5)] - pyspark.sql.utils.ParseException: 
FAILED ../../src/main/python/arithmetic_ops_test.py::test_int_division_mixed[Decimal(13,2)-Decimal(5,4)] - pyspark.sql.utils.ParseException: 
FAILED ../../src/main/python/arithmetic_ops_test.py::test_int_division_mixed[Decimal(13,2)-Decimal(3,-2)] - pyspark.sql.utils.ParseException:
FAILED ../../src/main/python/arithmetic_ops_test.py::test_cast_neg_to_decimal_err - AssertionError
 


FAILED ../../src/main/python/window_function_test.py::test_window_aggs_for_rows[[('a', RepeatSeq(Long)), ('b', Decimal(23,10)), ('c', Decimal(23,10))]-1000][IGNORE_ORDER({'local': True})] - AssertionError: GPU and CPU decimal values are different at [0, 'avg_c']
FAILED ../../src/main/python/window_function_test.py::test_window_aggs_for_rows[[('a', RepeatSeq(Long)), ('b', Decimal(23,10)), ('c', Decimal(23,10))]-1g][IGNORE_ORDER({'local': True})] - AssertionError: GPU and CPU decimal values are different at [0, 'avg_c']
FAILED ../../src/main/python/window_function_test.py::test_window_aggregations_for_decimal_ranges[[('a', RepeatSeq(Long)), ('b', Decimal(38,2)), ('c', Decimal(38,2))]][IGNORE_ORDER({'local': True})] - AssertionError: GPU and CPU int values are different at [0, 'count(1) OVER (PARTITION BY a ORDER BY b ASC NULLS FIRST RANGE BETWEEN (- 10.2345) FOLLOW...
FAILED ../../src/main/python/window_function_test.py::test_window_aggregations_for_big_decimal_ranges[[('a', RepeatSeq(Long)), ('b', Decimal(38,2)), ('c', Decimal(38,2))]][IGNORE_ORDER({'local': True})] - AssertionError: GPU and CPU int values are different at [0, 'count(1) OVER (PARTITION BY a ORDER BY b ASC NULLS FIRST RANGE BETWEEN (- 123456789012345...


```


**Steps/Code to reproduce bug**
Run integration tests for failing tests
cd  spark-rapids/integration_tests/
`./run_pyspark_from_build.sh -k <name of test>`       `//Example : ./run_pyspark_from_build.sh -k test_day_time_interval_divided_by_zero`


**Expected behavior**
Tests should pass OR XFAIL if there is a valid reason.

**Environment details (please complete the following information)**
 local

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala
@@ -961,6 +961,7 @@ object DataTypeMeta {
       Some(expr.dataType)
     } catch {
       case _: java.lang.UnsupportedOperationException => None
+      case _: org.apache.spark.SparkException => None
     }
     new DataTypeMeta(wrapped, overrideType)
   }

